File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int st =0;
        int end = numbers.size()-1;
        while(st<end){
            int cursum = numbers[st] + numbers[end];
            if(cursum == target){
                return {st+1, end+1};
            }
            else if(cursum<target){
                st++;
            }
            else{
                end--;
            }
        }
        return {};
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: list[int], target: int) -> list[int]:
        start =0
        end =len(numbers)-1
        while start<end:
            curSum = numbers[start] + numbers[end]
            if curSum == target:
                return [start+1,end+1]
            elif curSum <target:
                start+=1
            else:
                end -= 1